predicators/envs/vlm_envs.py

```diff
@@ -0,0 +1,147 @@
+
+Will likely be updated and potentially split into separate files in the
+future.
+"""
+
+import abc
+from typing import List, Optional, Sequence, Set
+
+import matplotlib
+import numpy as np
+from gym.spaces import Box
+
+from predicators.envs import BaseEnv
+from predicators.settings import CFG
+from predicators.structs import Action, EnvironmentTask, GroundAtom, Object, \
+    Predicate, State, Type
+
+
+class VLMPredicateEnv(BaseEnv):
+    """Environments that use VLM Predicates.
+
+    Note that no simulate function or ground truth model is implemented
+    for these yet. These are forthcoming.
+    """
+
+    def __init__(self, use_gui: bool = True) -> None:
+        super().__init__(use_gui)
+
+        # Types
+        self._object_type = Type("object", [])
+        self._goal_object_type = Type("goal_object", ["goal_true"],
+                                      self._object_type)
+
+        # Predicates
+        self._DummyGoal = Predicate("DummyGoal", [self._goal_object_type],
+                                    self._Dummy_Goal_holds)
+
+    def simulate(self, state: State, action: Action) -> State:
+        raise ValueError("simulate shouldn't be getting called!")
+
+    @property
+    def types(self) -> Set[Type]:
+        return {self._object_type, self._goal_object_type}
+
+    def _Dummy_Goal_holds(self, state: State,
+                          objects: Sequence[Object]) -> bool:
+        obj, = objects
+        return state.get(obj, "goal_true") > 0.5
+
+    def _generate_train_tasks(self) -> List[EnvironmentTask]:
+        return self._get_tasks(num=CFG.num_train_tasks, rng=self._train_rng)
+
+    def _generate_test_tasks(self) -> List[EnvironmentTask]:
+        return self._get_tasks(num=CFG.num_test_tasks, rng=self._test_rng)
+
+    @property
+    def predicates(self) -> Set[Predicate]:
+        return set([self._DummyGoal])
+
+    @property
+    def goal_predicates(self) -> Set[Predicate]:
+        return set([self._DummyGoal])
+
+    @property
+    def action_space(self) -> Box:
+        return Box(low=0.0, high=1.0, shape=(0, ), dtype=np.float32)
+
+    def render_state_plt(
+            self,
+            state: State,
+            task: EnvironmentTask,
+            action: Optional[Action] = None,
+            caption: Optional[str] = None) -> matplotlib.figure.Figure:
+        raise ValueError("shouldn't be trying to render env at any point!")
+
+    @property
+    @abc.abstractmethod
+    def get_vlm_debug_atom_strs(self) -> Set[str]:
+        """Return a set of atom strings that should be sufficient for a VLM to
+        label demonstrations consistently to learn good operators."""
+        raise NotImplementedError(
+            "VLM debug atom strings not implemented for this environment.")
+
+    def _get_tasks(
+        self, num: int, rng: np.random.Generator
+    ) -> List[EnvironmentTask]:  # pragma: no cover.
+        del num, rng
+        raise NotImplementedError("Override!")
```

Comment:
can this method just be removed, if this environment is meant to still be ABC?

Reply:
I did that originally, but the linter started complaining :/